LIPS/lip-17.md

```diff
@@ -0,0 +1,350 @@
+
+## Simple Summary
+
+The on-chain relays whitelist is planned to be used by Node Operators participating in the Lido protocol after the Merge to extract MEV according to the expected Lido policies.
+
+## Motivation
+
+It's proposed that Node Operators should use [`MEV-Boost`](https://github.com/flashbots/mev-boost) infrastructure developed by Flashbots to support MEV extraction through the open market mechanics as a current PBS solution that has a market fit.
+
+The proposed whitelist is intended to be a source of truth for the set of possible relays allowed to be used by Node Operators. In particular, Node Operators would use the contract to keep their software configuration up-to-date (setting the necessary relays once Lido DAO updates the set).
+
+## Mechanics
+
+The contract represents simple registry storage.
+
+Anyone can access the storage in a permissionless way through the disclosed `view` methods or by collecting the emitted events covering all storage modifications.
+
+Any modification of the set (i.e., internal storage modification) is allowed only by [the general Lido DAO governance process](https://lido.fi/governance#regular-process), which is implemented on-chain through the Aragon voting from the initial deployment stage. However, it's still possible to assign a dedicated management entity for adding/removing relay items.
+
+The proposed `MEVBoostRelayWhitelist` contract is non-upgradable, and its immutable owner is intended to be initialized with the [Lido DAO Aragon Agent](https://etherscan.io/address/0x3e40D73EB977Dc6a537aF587D48316feE66E9C8c) address on mainnet upon the deployment phase.
+
+The dedicated management `manager` entity is initialized with a zero address and can be set later by the contract's owner to be able to add/remove relays. The possible candidate to bear such responsibilities without sacrificing governance security is [Easy Track](https://easytrack.lido.fi/) (in this case  [`EVMScriptExecutor (Easy Track)`](https://docs.lido.fi/deployed-contracts#easy-track) should be set as `manager`).
+
+### Relay information structure
+
+It's proposed to use the following structure:
+
+```vyper
+struct Relay:
+    uri: String[MAX_STRING_LENGTH]
+    operator: String[MAX_STRING_LENGTH]
+    is_mandatory: bool
+    description: String[MAX_STRING_LENGTH]
+```
+
+where:
+
+- `uri` is the relay's URI to fetch the data
+- `operator` is the relay title or pseudonym
+- `is_mandatory` is supposed to distinguish between optional and mandatory relays
+- `description` is designated to store any additional info (unspecified at the current stage)
+
+### Adding relay
+
+To add a single new relay, `Lido DAO Aragon Agent` or `manager` (if assigned) should call `add_relay` method, passing all necessary relay structure's params described above.
+
+### Removing relay
+
+To add a single new relay, `Lido DAO Aragon Agent` or `manager` (if assigned) should call `remove_relay` method, passing the relay's `uri`.
```

Comment:
... --> To remove

I would also omit "single": it makes me expect description of a method to add multiple ones

Reply:
Confirm :+1: